### ISSUE TITLE
Get rid of methods

### ISSUE THREAD

**Issue by schanzer** [bootstrapworld/curriculum#1087]
- [x] Remove all references to methods, and change method names in all files
- [x] Consider re-ordering lessons
- [x] move `sort` to an earlier lesson?
- [x] `table-functions` - new lesson replacing `table-methods` which covers the same content in (possibly) less time. Use with Circles of Evaluation
- [x] `method-chaining` - deal with this last. Likely fold this into `table-functions` or `grouped-samples`
- [x] Talk with Rachel about overlaps with data-science-codap

-------------------------------------
What follows is a pretty substantial shift for Data Science -- it would only be rolled out for 2023, but it has implications for some other projects on our plate.

**Background (read this first!)**
Right now Bootstrap:Data Science uses methods - a _lot_. This has two specific benefits:

1. When chaining methods, the operations are listed in order. `table.filter(x).filter(y)` will use `x` before `y`. This makes the order of operations clear and improves readability.
2. Methods are a computer-sciency thing that _other people_ think is fancy and important. It's pure marketing to the CS crowd.

Using methods comes at a price, however:

1. More syntax -- students need to learn about dot accessors, which introduce Yet Another Thing for teachers to spend time on
2. More complex contracts -- the method contracts look very different from function contracts, making them harder to read and introducing Yet Another Thing for teachers to spend time on.
3. More _semantics_ -- students need to build a more sophisticated mental model to understand that methods somehow _live inside_ tables, and that `A.row-n(0)` and `B.row-n(0)` will produce different results. We already spend a little time on this in our lessons and PD, explaining that `row-n` can be thought of as a function with a "secret argument", which is the table itself. 
4. Less re-use -- we already introduce `image-scatter-plot` as an intermediate step on the way to Table Methods. This function introduces the idea that a function can take _other functions_ as arguments. But that only gets us halfway there! We then spend an entire lesson teaching students about methods, then explaining that methods can ALSO take functions as arguments.

We switched to methods back in 2018, shying away from special forms like `sieve` and `extend` which require lots more syntax. We truly gave special forms a try, and problems 1 and 2 (above) were so bad that we had to junk the whole thing.

But there IS another way...

**Proposal: What if we used functions for _everything?_**

So instead of `animals-table.filter(is-cat)`, we would write something like `filter(animals-table, is-cat)`.

There quite a few benefits to this approach, aligned to the costs of methods:

1. Syntax -- filtering a table _looks_ the same as making a star
2. Contracts -- the contract for filter-table is just `filter:: (t::Table, test::Row->Boolean) -> Table`, which looks like any other contract
3. Semantics -- the only mental model required is standard function composition
4. Re-use -- know how to draw a star? Cool, then you already know how to filter a table

The primary cost that worries me is _readability_. Function composition works backwards: `f(g(14))` applies the function `g` before applying `f`. Compare the two (identical) expressions below:

_Methods_
```
animals-table.filter(is-cat).filter(is-fixed).order-by("age", true").build-column("label", sticker)
```

_Functions_
```
build-column(order-by(filter(filter(animals-table, is-cat), is-fixed), "age", true"), "label", sticker)
```
I find the second one pretty hard to read, even if it _is_ standard function notation. 

There are, however, a few other benefits:

1. This would also allow us to use the Circles of Evaluation with tables! Talk about re-use...

![image](https://user-images.githubusercontent.com/77126/183297151-b9dbc85f-33f1-4504-8ec7-d502d4585ae1.png)

2. This would provide a lot more alignment with CODAP, since composing Transformers creates tables that are named using function notation. This will make Rachel's life a little easier, and may open up more opportunities to re-use material and pedagogy across both DS pathways.

3. The Table-Methods lesson becomes extremely lightweight, with virtually no work spent on syntax, and the Method-Chaining lesson disappears completely!

Basically it's "readability of chained methods" v. every other benefit. So....what do you all think? How important is that readability factor?
